### DISCUSSION TITLE
response_model_by_alias should be passed down to the Pydantic.dict() method

### DISCUSSION THREAD

**Discussion by frco9** [fastapi/fastapi#8672] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def _prepare_response_content(
    res: Any,
    *,
    exclude_unset: bool,
    exclude_defaults: bool = False,
    exclude_none: bool = False,
) -> Any:
    if isinstance(res, BaseModel):
        read_with_orm_mode = getattr(res.__config__, "read_with_orm_mode", None)
        if read_with_orm_mode:
            # Let from_orm extract the data from this model instead of converting
            # it now to a dict.
            # Otherwise there's no way to extract lazy data that requires attribute
            # access instead of dict iteration, e.g. lazy relationships.
            return res
        return res.dict(
            by_alias=True,
            exclude_unset=exclude_unset,
            exclude_defaults=exclude_defaults,
            exclude_none=exclude_none,
        )

(....)
```

### Description

In the `router.py` response content method, the `by_alias` ([here](https://github.com/tiangolo/fastapi/blob/20d48345465e5b09756cc9ef62fb940c63255a13/fastapi/routing.py#L76)) parameter is set to True when converting a Pydantic model to dict while every other options is overridable using a parameter like [response_model_exclude_unset](https://fastapi.tiangolo.com/tutorial/response-model/?h=response_model_exclude_unset#use-the-response_model_exclude_unset-parameter). 
This is an issue if we want to only use a Pydantic model alias for input data but not as output. Thus we should be able to override this default param. 

### Wanted Solution

Use the existing `response_model_by_alias` param and pass down its value to the `_prepare_response_content` method. 

### Wanted Code

```python
def _prepare_response_content(
    res: Any,
    *,
    exclude_unset: bool,
    exclude_defaults: bool = False,
    exclude_none: bool = False,
    by_alias: bool = True,
) -> Any:
    if isinstance(res, BaseModel):
        read_with_orm_mode = getattr(res.__config__, "read_with_orm_mode", None)
        if read_with_orm_mode:
            # Let from_orm extract the data from this model instead of converting
            # it now to a dict.
            # Otherwise there's no way to extract lazy data that requires attribute
            # access instead of dict iteration, e.g. lazy relationships.
            return res
        return res.dict(
            by_alias=by_alias,
            exclude_unset=exclude_unset,
            exclude_defaults=exclude_defaults,
            exclude_none=exclude_none,
        )

(....)
```

### Operating System

macOS

### FastAPI Version

0.70.0

### Python Version

3.9.7

**Comment by YuriiMotov** [comment, marked as answer]
It's unlikely to be implemented since this would make the openapi schema invalid.

In Pydantic V2 use `validation_alias` when you need to use alias for validation only
